src/main/java/net/openhft/chronicle/wire/VanillaFieldInfo.java

```diff
@@ -18,111 +18,98 @@
 package net.openhft.chronicle.wire;
 
 import net.openhft.chronicle.core.Jvm;
+import net.openhft.chronicle.core.UnsafeMemory;
 import net.openhft.chronicle.core.util.ObjectUtils;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
 import java.lang.reflect.Field;
 import java.lang.reflect.ParameterizedType;
 import java.lang.reflect.Type;
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.List;
 import java.util.Objects;
-import java.util.stream.Collectors;
-
-import static net.openhft.chronicle.wire.WireMarshaller.WIRE_MARSHALLER_CL;
 
+/**
+ * @deprecated This class will become internal in the future, use {@link FieldInfo#createForField(String, Class, BracketType, Field)} instead.
+ */
+@Deprecated
 @SuppressWarnings("rawtypes")
 public class VanillaFieldInfo extends AbstractFieldInfo implements FieldInfo {
 
+    private static final long UNSET_OFFSET = Long.MAX_VALUE;
+
     private final Class parent;
     private transient Field field;
+    private transient long offset = UNSET_OFFSET;
```

Comment:
If we go through a serialize/deserialize cycle it seems prudent to refresh the offset as we may be on different JVM or have a different implementation of the class available on the classpath.